### PR TITLE
[core] improvements

### PR DESCRIPTION
**Pull request by AndrewShkrob**
* Make dp::Color constexpr
* Fix warning in `visual_params.hpp`

### CODE REVIEW

File: drape/color.hpp [hunk 1]
```diff
@@ -6,63 +6,70 @@
 #include <sstream>
 #include <string>
 
+#define EXTRACT_BYTE(x, n) (static_cast<uint8_t>(((x) >> 8 * (n)) & 0xFF))
+#define CHANNEL_TO_FLOAT(x) ((x) / 255.0 /* MaxChannelValue */)
```
Comment (vng) [inline drape/color.hpp:10]: static constexpr functions inside Color struct scope?

File: drape/color.hpp [hunk 2]
```diff
@@ -6,63 +6,70 @@
...
+
 namespace dp
 {
 struct Color
 {
-  Color();
-  explicit Color(uint32_t rgba);
-  Color(uint8_t red, uint8_t green, uint8_t blue, uint8_t alpha);
+  constexpr Color() : m_rgba(255 /* MaxChannelValue */) {}
```
Comment (vng) [inline drape/color.hpp:16]: Write a comment, what default color actually is. What is "MaxChannelValue"? Is it (0, 0, 0, alfa = 255) ?

**Review by vng — APPROVED** [review]

File: drape/color.hpp [hunk 3]
```diff
@@ -10,59 +10,68 @@ namespace dp
...
-  static Color Green() { return Color(0, 255, 0, 255); }
-  static Color Yellow() { return Color(255, 255, 0, 255); }
-  static Color Transparent() { return Color(0, 0, 0, 0); }
+  constexpr static Color FromARGB(uint32_t argb)
+  {
+    return {ExtractByte(argb, 2), ExtractByte(argb, 1), ExtractByte(argb, 0), ExtractByte(argb, 3)};
+  }
 
 private:
+  constexpr static uint8_t ExtractByte(uint32_t number, uint8_t byteIdx) { return ((number) >> 8 * (byteIdx)) & 0xFF; }
```
Comment (biodranik) [inline drape/color.hpp:59]:
```suggestion
  constexpr static uint8_t ExtractByte(uint32_t number, uint8_t byteIdx) { return (number >> (8 * byteIdx)) & 0xFF; }
```

**Review by biodranik — APPROVED** [review]
Cool, thanks! It would be great to do the same for points and rects (at least zero ones/invalid ones).

File: drape/color.hpp (hunk 3 above)
Comment (AndrewShkrob) [reply drape/color.hpp]: done

**Review by biodranik — APPROVED** [review]
Thanks!

**Pull request merged**